exchange/replication.go

```diff
@@ -423,11 +425,18 @@ func (r *Replication) handleRequest(s network.Stream) {
 					log.Error().Err(err).Msg("error when fetching keys")
 				}
 
-				if err := r.idx.SetRef(&DataRef{
+				ref := &DataRef{
 					PayloadCID:  req.PayloadCID,
 					PayloadSize: int64(req.Size),
 					Keys:        keys.AsBytes(),
-				}); err != nil {
+				}
+
+				err = r.idx.SetRef(ref)
+				if err == ErrRefAlreadyExists {
```

Comment:
I don't believe it's possible the ref already exists since we check and return before no?